### PR TITLE
feat: Support HTTPS endpoints

### PR DESCRIPTION
**Pull request by ifindlay-cci**
### Checklist:

* [ ] Have you added or updated tests to validate the changed functionality?
* [ ] Have you added Release Notes in the docs repositories?
* [ ] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: internal/testdrive/broker_start.go [hunk 1]
```diff
@@ -83,7 +94,16 @@ func StartBroker(csbPath, bpk, db string, opts ...StartBrokerOption) (*Broker, e
 
 	start := time.Now()
 	for {
-		response, err := http.Head(fmt.Sprintf("http://localhost:%d", port))
+		scheme := "http"
+		for _, envVar := range cmd.Env {
+			if strings.HasPrefix(envVar, "TLS_") {
+				http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```
Comment (zucchinidev) [inline internal/testdrive/broker_start.go:100]: issue: Avoid modifying the global `http.DefaultTransport` directly. Instead, create a custom transport.

File: internal/testdrive/broker_start.go [hunk 2]
```diff
@@ -83,7 +94,16 @@ func StartBroker(csbPath, bpk, db string, opts ...StartBrokerOption) (*Broker, e
 
 	start := time.Now()
 	for {
```
Comment (zucchinidev) [inline internal/testdrive/broker_start.go:96]:
```suggestion
	scheme := "http"
	transport := &http.Transport{}
	for _, envVar := range cmd.Env {
		if strings.HasPrefix(envVar, "TLS_") {
			http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
			scheme = "https"
			break
		}
	}

	for {
		httpClient := &http.Client{Transport: transport}
		response, err := httpClient.Head(fmt.Sprintf("%s://localhost:%d", scheme, port))
```

File: internal/testdrive/broker_start.go [hunk 3]
```diff
@@ -99,6 +119,95 @@ func StartBroker(csbPath, bpk, db string, opts ...StartBrokerOption) (*Broker, e
...
+		Expect(err).NotTo(HaveOccurred())
+		defer certFileBuf.Close()
+
+		privKeyFileBuf, err := os.CreateTemp("", "")
+		Expect(err).NotTo(HaveOccurred())
+		defer privKeyFileBuf.Close()
+
+		if !isValid {
+			serverPrivKey[10] = 'a'
+		}
```
Comment (zucchinidev) [inline internal/testdrive/broker_start.go:200]:
```suggestion
		if !isValid {
		       // If the isValid parameter is false, the server private key is intentionally corrupted
		       // by modifying one of its bytes.
			serverPrivKey[10] = 'a'
		}
```

File: internal/testdrive/broker_start.go [hunk 4]
```diff
@@ -99,6 +119,95 @@ func StartBroker(csbPath, bpk, db string, opts ...StartBrokerOption) (*Broker, e
...
+		BasicConstraintsValid: true,
+	}
+
+	caPrivKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	Expect(err).NotTo(HaveOccurred())
+
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPrivKey.PublicKey, caPrivKey)
+	Expect(err).NotTo(HaveOccurred())
+
+	caPEM, _ := encodeKeyPair(caBytes, x509.MarshalPKCS1PrivateKey(caPrivKey))
```
Comment (zucchinidev) [inline internal/testdrive/broker_start.go:141]: issue (non-blocking): the PEM-encoded CA certificate is not being used. The code is not necessary, by removing it, we will improve the readability and the public API of the createCAKeyPair function

File: internal/testdrive/broker_start.go [hunk 5]
```diff
@@ -99,6 +119,95 @@ func StartBroker(csbPath, bpk, db string, opts ...StartBrokerOption) (*Broker, e
...
+	}
+
+	certPrivKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	Expect(err).NotTo(HaveOccurred())
+
+	certBytes, err := x509.CreateCertificate(rand.Reader, cert, ca, &certPrivKey.PublicKey, caPrivKey)
+	Expect(err).NotTo(HaveOccurred())
+
+	certPEM, certPrivKeyPEM := encodeKeyPair(certBytes, x509.MarshalPKCS1PrivateKey(certPrivKey))
+	return certPEM, certPrivKeyPEM
```
Comment (zucchinidev) [inline internal/testdrive/broker_start.go:167]: thoughts: I prefer to declare the variable using the semantic value and return it with the semantic pointer. If the body of the functions grows, the readability gets worse.

File: cmd/serve.go [hunk 6]
```diff
@@ -210,7 +214,26 @@ func startServer(registry pakBroker.BrokerRegistry, db *sql.DB, brokerapi http.H
 	port := viper.GetString(apiPortProp)
 	host := viper.GetString(apiHostProp)
 	logger.Info("Serving", lager.Data{"port": port})
-	_ = http.ListenAndServe(fmt.Sprintf("%s:%s", host, port), router)
+
+	tlsCertCaBundle := viper.GetString(tlsCertCaBundleProp)
+	tlsKey := viper.GetString(tlsKeyProp)
+
+	logger.Info("tlsCertCaBundle", lager.Data{"tlsCertCaBundle": tlsCertCaBundle})
+	logger.Info("tlsKey", lager.Data{"tlsKey": tlsKey})
```
Comment (zucchinidev) [inline cmd/serve.go:222]: issue: we want to avoid logging sensitive data.

File: cmd/serve.go [hunk 7]
```diff
@@ -210,7 +214,26 @@ func startServer(registry pakBroker.BrokerRegistry, db *sql.DB, brokerapi http.H
...
+		Handler: router,
+	}
+
+	if tlsCertCaBundle != "" && tlsKey != "" {
+		err := httpServer.ListenAndServeTLS(tlsCertCaBundle, tlsKey)
+		if err != nil {
+			logger.Fatal("Failed to start broker", err)
+		}
+	} else {
+		_ = httpServer.ListenAndServe()
```
Comment (zucchinidev) [inline cmd/serve.go:235]:
```suggestion
		err := httpServer.ListenAndServe()
	        if err != nil {
		        logger.Fatal("Failed to start broker without TLS", err)
	        }
```

**Review by zucchinidev — CHANGES_REQUESTED** [review]

File: internal/testdrive/broker_start.go (hunk 1 above)
Comment (nouseforaname) [reply internal/testdrive/broker_start.go:100]: for tests?

[the remark this answers is not in the document]
Comment (nouseforaname) [reply internal/testdrive/broker_start.go:100]: again, for tests?

File: cmd/serve.go (hunk 6 above)
Comment (nouseforaname) [reply cmd/serve.go:222]: thats a path

File: cmd/serve.go (hunk 7 above)
Comment (nouseforaname) [reply cmd/serve.go:235]: we cannot fatal here, as ListenAndServe always returns a non nil error

File: internal/testdrive/broker_start.go (hunk 5 above)
Comment (nouseforaname) [reply internal/testdrive/broker_start.go:167]: I'm not sure I understand how the readability will get better by returning a pointer?

**Pull request merged**